### PR TITLE
fix(DatePicker): optimize RangePicker's range area style

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Before:

image

After:

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | optimize RangePicker's range area style |
| 🇨🇳 Chinese | 优化 RangePicker 选择区域样式  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 956bc86</samp>

This pull request improves the style and code quality of the date picker component. It adjusts the border radius of the range picker cells to match the design spec, and simplifies the object assignment in `generateRangePicker.tsx`.

### <samp>🤖 Generated by Copilot at 956bc86</samp>

*  Simplify the assignment of `additionalOverrideProps` in `generateRangePicker.tsx` by using a single object literal ([link](https://github.com/ant-design/ant-design/pull/44206/files?diff=unified&w=0#diff-fba17c237c150271f91ee2ca848654484ecbbbcc71b010a8947620e945e10e34L71-R71))
*  Update the border radius values of the inner cell style for the range picker component in `index.ts` to match the design spec and use `borderRadiusSM` instead of `lineWidth` ([link](https://github.com/ant-design/ant-design/pull/44206/files?diff=unified&w=0#diff-a4f33cb3f0dfef43aa860e509b2c43fc62168efd797b258ed2cf47dfc4b1eaa4L266-R267), [link](https://github.com/ant-design/ant-design/pull/44206/files?diff=unified&w=0#diff-a4f33cb3f0dfef43aa860e509b2c43fc62168efd797b258ed2cf47dfc4b1eaa4L278-R279))

### CODE REVIEW

**Comment by afc163** [comment]
图片

右边接缝处最好是方的。

**Comment by kiner-tang** [comment]
> 图片
> 右边接缝处最好是方的。

👌🏻

File: components/date-picker/generatePicker/generateRangePicker.tsx [hunk 1]
```diff
@@ -68,9 +68,7 @@ export default function generateRangePicker<DateType>(generateConfig: GenerateCo
 
     const [wrapSSR, hashId] = useStyle(prefixCls);
 
-    let additionalOverrideProps: any = {};
-    additionalOverrideProps = {
-      ...additionalOverrideProps,
```
Comment (MadCcc) [inline components/date-picker/generatePicker/generateRangePicker.tsx:73]: 这个去掉是为啥

File: components/date-picker/style/index.ts [hunk 2]
```diff
@@ -130,6 +130,12 @@ const genPickerCellInnerStyle = (token: SharedPickerToken): CSSObject => {
       borderRadius: borderRadiusSM,
       transition: `background ${motionDurationMid}, border ${motionDurationMid}`,
     },
+    [`&-range-hover-start, &-range-hover-end`]: {
+      [pickerCellInnerCls]: {
+        borderStartEndRadius: 0,
+        borderEndEndRadius: 0,
+      },
+    },
```
Comment (MadCcc) [inline components/date-picker/style/index.ts:138]: 这个不加会咋样

Comment (kiner-tang) [reply components/date-picker/style/index.ts:138]: image
会这样

Comment (MadCcc) [reply components/date-picker/style/index.ts:138]: 这里本来也是方的吧

Comment (MadCcc) [reply components/date-picker/style/index.ts:138]: 应该是圆角导致周长变化了，所以会不对齐

Comment (kiner-tang) [reply components/date-picker/style/index.ts:138]: 不是的，这是正式环境官网的截图，也是有圆角的：
image

Comment (kiner-tang) [reply components/date-picker/style/index.ts:138]: 但实际上我们在已经选择的时候，也是有圆角的呀，如果因为圆角导致对不齐，那是不是这个也有问题呢？
image
而且我好像没看到哪里没对齐

**Review by MadCcc — APPROVED** [review]

**Pull request merged**